webpack/ForemanTasks/Components/TaskDetails/TaskDetailsReducer.js

```diff
@@ -1,42 +1,20 @@
 import Immutable from 'seamless-immutable';
-import { STATUS } from 'foremanReact/constants';
 import {
-  FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS,
   FOREMAN_TASK_DETAILS_STOP_POLLING,
   FOREMAN_TASK_DETAILS_START_POLLING,
-  FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE,
+  FOREMAN_TASK_DETAILS_SUCCESS,
 } from './TaskDetailsConstants';
 
 const initialState = Immutable({ isData: false });
 
-export default (state = initialState, action) => {
-  const { type, payload } = action;
-  let { taskReload, timeoutId } = state;
-
+export default (state = initialState, { type }) => {
   switch (type) {
-    case FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST:
-      return state.set('status', STATUS.PENDING);
-    case FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS:
-      if (payload.state === 'stopped') {
-        clearTimeout(state.timeoutId);
-        taskReload = false;
-        timeoutId = null;
-      }
-      return state.merge({
-        status: STATUS.RESOLVED,
-        isData: true,
-        ...payload,
-        taskReload,
-        timeoutId,
-      });
-    case FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE:
-      return state.merge({ status: STATUS.ERROR, error: payload });
+    case FOREMAN_TASK_DETAILS_SUCCESS:
+      return state.set('isData', true);
```

Comment:
why do you need also a local state to check if there's data?
not sure if you really need it (saw it is being used for the loading indicator)
if you must use it, you can use a selector on the API response.

Reply:
This is from the previous PR, its used because there is an option for the data to be refreshed, so the first time the component gets the data it will show it and won't go into loading state each time the is a re-fetching of the data.
First tasks request -> show loading
success -> show data
second request -> show data from previous request
second success -> show new data

Reply:
Can you use a selector for it from the API ?
As the response will appear only after success/error, no need for the local state for this logic IMO

Reply:
No, because it shouldn't change if the status changes, is should only change in the first time there is FOREMAN_TASK_DETAILS_SUCCESS

Reply:
Also all of the data is optional so I can't check if I have data

Reply:
thanks @MariaAga , lets keep it than